userspace/libsinsp/sinsp_filtercheck_thread.cpp

```diff
@@ -761,21 +825,104 @@ uint8_t* sinsp_filter_check_thread::extract(sinsp_evt *evt, OUT uint32_t* len, b
 	case TYPE_ENV:
 		{
 			m_tstr.clear();
-
 			uint32_t j;
 			const auto& env = tinfo->get_env();
 			uint32_t nargs = (uint32_t)env.size();
 
-			for(j = 0; j < nargs; j++)
+			if(nargs == 0)
 			{
-				m_tstr += env[j];
-				if(j < nargs -1)
+				RETURN_EXTRACT_STRING(m_tstr);
+			}
+
+			if(!m_argname.empty()) // contains ENV_NAME we are looking to extract
+			{
+				// proc.env[ENV_NAME] use case: returns matched env variable value
+				for (const auto& item : env)
 				{
-					m_tstr += ' ';
+					std::string str(item.data(), item.data() + item.size());
```

Comment:
`item` should already be a string in the form of `ENV_NAME=env_value` or am I missing something?

Reply:
Adjusted the approach -- after looking at it a map https://github.com/falcosecurity/libs/pull/1561#discussion_r1428131112 would have been extra not necessary work and the way env is stored and processed throughout in libsinsp is (as many things) very intertwined.